python/gtsam/examples/TranslationAveragingExample.py

```diff
@@ -0,0 +1,122 @@
+    """Estimate poses given rotations normalized translation directions between cameras.
+
+    Args:
+        relative_translations -- List of normalized translation directions between camera pairs, each direction
+                                 is from the first camera to the second, in the frame of the first camera.
+        rotations -- Rotations of the cameras in the world frame.
+
+    Returns:
+        Values -- Estimated poses.
+    """
+
+    # Some hyperparameters, values used from 1dsfm.
+    max_1dsfm_projection_directions = 48
+    outlier_weight_threshold = 0.1
+
+    # Convert the translation directions to global frame using the rotations.
+    w_relative_translations = gtsam.BinaryMeasurementsUnit3()
+    for relative_translation in relative_translations:
+        w_relative_translation = gtsam.Unit3(rotations.atRot3(relative_translation.key1())
+                                             .rotate(relative_translation.measured().point3()))
+        w_relative_translations.append(gtsam.BinaryMeasurementUnit3(relative_translation.key1(),
+                                                                    relative_translation.key2(),
+                                                                    w_relative_translation,
+                                                                    relative_translation.noiseModel()))
+
+    # Indices of measurements that are to be used as projection directions. These are randomly chosen.
+    sampled_indices = np.random.choice(len(w_relative_translations), min(
+        max_1dsfm_projection_directions, len(w_relative_translations)), replace=False)
+    # Sample projection directions from the measurements.
+    projection_directions = [
+        w_relative_translations[idx].measured() for idx in sampled_indices]
+
+    outlier_weights = []
+    # Find the outlier weights for each direction using MFAS.
+    for direction in projection_directions:
+        algorithm = gtsam.MFAS(w_relative_translations, direction)
+        outlier_weights.append(algorithm.computeOutlierWeights())
+
+    # Compute average of outlier weights. Each outlier weight is a map from a pair of Keys (camera IDs) to a weight,
+    # where weights are proportional to the probability of the edge being an outlier.
+    avg_outlier_weights = defaultdict(lambda: 0.0)
+    for outlier_weight_dict in outlier_weights:
+        for keypair, weight in outlier_weight_dict.items():
+            avg_outlier_weights[keypair] += weight / len(outlier_weights)
+
+    # Remove w_relative_tranlsations that have weight greater than threshold, these are outliers.
+    inlier_w_relative_translations = gtsam.BinaryMeasurementsUnit3()
+    [inlier_w_relative_translations.append(Z) for Z in w_relative_translations
+        if avg_outlier_weights[(Z.key1(), Z.key2())] < outlier_weight_threshold]
+
+    # Run the optimizer to obtain translations for normalized directions.
+    w_translations = gtsam.TranslationRecovery(
```

Comment:
I'm still a bit hesitant about our notations here. My preference is too be overly explicit, like this: https://github.com/argoai/argoverse-api#a-note-regarding-coordinate-transforms

Reply:
I understand the concern, at the same time, I think the notation that you point to `p_dst = dst_SE3_src * p_src` might contradict with the GTSAM convention (if I am not wrong). Ideally I think we should prefer `dst_p = dst_SE3_src * src_p`. Is there a variable name you recommend here? I called them translations to be somewhat consistent in the file (we have relative_translations), but `w_T_c` would work as well.

Reply:
i'm good with either of those. I just get very confused when I read only `rotation` or `pose`. I would much prefer if we use something like the following throughout this whole file:
`wTc` for SE(3) transformations from camera to world
`wtc` for translations from camera to world
`wRc` for rotations from camera to world